### PR TITLE
BLOCK_HIST capability enabled if archive_mode=true

### LINKED ISSUE #3562: Add BLOCK_HIST capability and enable it if archive_mode=true
We want to advertise an additional `BLOCK_HIST` capability if we are running with `archive_mode=true`.
This will enable other peers to request historical blocks from us along with other peers that advertise support for this.

Related #3552.

### PR DESCRIPTION
**Pull request by antiochp**
Resolves #3562

### CODE REVIEW

File: p2p/tests/ser_deser.rs [hunk 1]
```diff
@@ -81,6 +81,6 @@ fn test_capabilities() {
 fn test_pibd_capabilities() {
 	assert_eq!(
 		p2p::types::Capabilities::default() | p2p::types::Capabilities::PIBD_HIST,
-		p2p::types::Capabilities::from_bits_truncate(0b11111111 as u32),
+		p2p::types::Capabilities::from_bits_truncate(0b00011111 as u32),
```
Comment (phyro) [inline p2p/tests/ser_deser.rs:84]: If I understand it correctly, the above value does an OR of default+PIBD bitmask. If I'm reading it correctly, the default already includes PIBD_HIST https://github.com/mimblewimble/grin/blob/010730e6daaa268a8316cc702c51fcb408fa0a6d/p2p/src/types.rs#L406
So maybe we can remove the `| p2p::types::Capabilities::PIBD_HIST` and correct the comment above the test which tells us that it is not in the defaults.

Comment (antiochp) [reply p2p/tests/ser_deser.rs:84]: Oh good catch. 👍 
I'll update in a separate PR (unrelated to this).

Comment (antiochp) [reply p2p/tests/ser_deser.rs:84]: https://github.com/mimblewimble/grin/pull/3565

Comment (antiochp) [reply p2p/tests/ser_deser.rs:84]: I'll clean this PR up once https://github.com/mimblewimble/grin/pull/3565 is merged.

**Comment by phyro** [comment]
would it make sense to add an assertion that it does not contain the BLOCK_HIST capability here?https://github.com/mimblewimble/grin/blob/master/p2p/tests/capabilities.rs#L44
or maybe in ser_deser.rs. Anyway, this looks good to me and don't want to stall it

**Review by phyro — APPROVED** [review]

**Review by DavidBurkett — APPROVED** [review]

**Pull request merged**